beacon-chain/rpc/eth/light-client/helpers.go

```diff
@@ -0,0 +1,377 @@
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/blockchain"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/rpc/apimiddleware"
+	"github.com/prysmaticlabs/prysm/v4/beacon-chain/state"
+	fieldparams "github.com/prysmaticlabs/prysm/v4/config/fieldparams"
+	"github.com/prysmaticlabs/prysm/v4/config/params"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/interfaces"
+	"github.com/prysmaticlabs/prysm/v4/consensus-types/primitives"
+	v1 "github.com/prysmaticlabs/prysm/v4/proto/eth/v1"
+	v2 "github.com/prysmaticlabs/prysm/v4/proto/eth/v2"
+	"github.com/prysmaticlabs/prysm/v4/time/slots"
+)
+
+const (
+	nextSyncCommitteeBranchNumOfLeaves = 5
+)
+
+// CreateLightClientBootstrap - implements https://github.com/ethereum/consensus-specs/blob/3d235740e5f1e641d3b160c8688f26e7dc5a1894/specs/altair/light-client/full-node.md#create_light_client_bootstrap
+// def create_light_client_bootstrap(state: BeaconState) -> LightClientBootstrap:
+//
+//	assert compute_epoch_at_slot(state.slot) >= ALTAIR_FORK_EPOCH
+//	assert state.slot == state.latest_block_header.slot
+//
+//	return LightClientBootstrap(
+//	    header=BeaconBlockHeader(
+//	        slot=state.latest_block_header.slot,
+//	        proposer_index=state.latest_block_header.proposer_index,
+//	        parent_root=state.latest_block_header.parent_root,
+//	        state_root=hash_tree_root(state),
+//	        body_root=state.latest_block_header.body_root,
+//	    ),
+//	    current_sync_committee=state.current_sync_committee,
+//	    current_sync_committee_branch=compute_merkle_proof_for_state(state, CURRENT_SYNC_COMMITTEE_INDEX)
+//	)
+func CreateLightClientBootstrap(ctx context.Context, state state.BeaconState) (*LightClientBootstrap, error) {
+	// assert compute_epoch_at_slot(state.slot) >= ALTAIR_FORK_EPOCH
+	if slots.ToEpoch(state.Slot()) < params.BeaconConfig().AltairForkEpoch {
+		return nil, fmt.Errorf("invalid state slot %d", state.Slot())
+	}
+
+	// assert state.slot == state.latest_block_header.slot
+	if state.Slot() != state.LatestBlockHeader().Slot {
+		return nil, fmt.Errorf("invalid state slot %d", state.Slot())
+	}
+
+	// Prepare data
+	latestBlockHeader := state.LatestBlockHeader()
+
+	currentSyncCommittee, err := state.CurrentSyncCommittee()
+	if err != nil {
+		return nil, fmt.Errorf("could not get current sync committee %v", err)
+	}
+
+	currentSyncCommitteePubkeys := currentSyncCommittee.GetPubkeys()
+	committee := apimiddleware.SyncCommitteeJson{
+		Pubkeys:         make([]string, len(currentSyncCommitteePubkeys)),
+		AggregatePubkey: hexutil.Encode(currentSyncCommittee.GetAggregatePubkey()),
+	}
+	for i, pubkey := range currentSyncCommitteePubkeys {
+		committee.Pubkeys[i] = hexutil.Encode(pubkey)
+	}
+	currentSyncCommitteeProof, err := state.CurrentSyncCommitteeProof(ctx)
+	if err != nil {
+		return nil, fmt.Errorf("could not get current sync committee proof %v", err)
+	}
+
+	branch := make([]string, len(currentSyncCommitteeProof))
```

Comment:
Yes, this is true.  Normally, it is safer to use len (which is cheap as well) to avoid possible mem issue. 
But in this special case, we can use constant.  

Patched.